src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -161,56 +156,87 @@ public boolean link() throws IOException, InterruptedException {
                 "-target", "1.7", "-cp", dalvikSrcPath.toString(), "-bootclasspath", androidJar,
                 dalvikSrcPath.resolve("MainActivity.java").toString());
-        processRunner.runProcess("dalvikCompilation");
+        processResult = processRunner.runProcess("dalvikCompilation");
+        if (processResult != 0)
+            return false;
+
         ProcessRunner dx = new ProcessRunner(buildToolsPath.resolve("dx").toString(), "--dex",
                 "--output="+dalvikBinPath.resolve("classes.dex"),dalvikClassPath.toString());
-        dx.runProcess("DX");
+        processResult = dx.runProcess("DX");
+        if (processResult != 0)
+            return false;
 
         ProcessRunner aaptpackage = new ProcessRunner(aaptCmd, "package", "-f", "-m", "-F", unalignedApk,
         "-M", androidManifestPath.toString(), "-I", androidJar);
-        aaptpackage.runProcess("AAPT-package");
+        processResult = aaptpackage.runProcess("AAPT-package");
+        if (processResult != 0)
+            return false;
 
         ProcessRunner aaptAddClass = new ProcessRunner(aaptCmd, "add", unalignedApk,
                 "classes.dex");
-        aaptAddClass.runProcess("AAPT-add classes", dalvikBinPath.toFile());
+        processResult = aaptAddClass.runProcess("AAPT-add classes", dalvikBinPath.toFile());
+        if (processResult != 0)
+            return false;
+
         Path libPath = paths.getAppPath().resolve(projectConfiguration.getAppName());
         Path graalLibPath = dalvikLibArm64Path.resolve("libmygraal.so");
         Files.deleteIfExists(graalLibPath);
         Files.copy(libPath, graalLibPath);
         Path freetypeLibPath = dalvikLibArm64Path.resolve("libfreetype.so");
         Files.deleteIfExists(freetypeLibPath);
         Files.copy(fileDeps.getJavaFXSDKLibsPath().resolve("libfreetype.so"), freetypeLibPath);
+
         ProcessRunner aaptAddLibs = new ProcessRunner(aaptCmd, "add", unalignedApk,
                 "lib/arm64-v8a/libmygraal.so","lib/arm64-v8a/libfreetype.so" );
-        aaptAddLibs.runProcess("AAPT-add lib", dalvikPath.toFile());
+        processResult = aaptAddLibs.runProcess("AAPT-add lib", dalvikPath.toFile());
+        if (processResult != 0)
+            return false;
 
         ProcessRunner zipAlign = new ProcessRunner(buildToolsPath.resolve("zipalign").toString(), "-f", "4", unalignedApk, alignedApk);
-        zipAlign.runProcess("zipalign");
+        processResult = zipAlign.runProcess("zipalign");
+        if (processResult != 0)
+            return false;
+
         createDevelopKeystore();
-        return true;
+
+        ProcessRunner sign =  new ProcessRunner(buildToolsPath.resolve("apksigner").toString(), "sign", "--ks", 
+            paths.getGvmPath().resolve("debugkeystore.jks").toString(), "--ks-key-alias", "androiddebugkey", "--ks-pass", "pass:android", "--key-pass", "pass:android",  alignedApk);
+        processResult = sign.runProcess("sign");
```

Comment:
This step fails for me: Failed PKCS12 integrity checking. Didn't look deeper yet, but maybe this has to do with an invalid keychain?

Reply:
Was `debugkeystore.jks` generated using this version? 
I used to get that error if I don't run `mvn clean` before building with this version (because old `debugkeystore.jks` generation was utterly broken)

Reply:
That was the reason indeed -- I forgot we already created a dummy keystore before.